solidity/tsconfig.json

```diff
@@ -2,7 +2,7 @@
   "extends": "../tsconfig.json",
   "compilerOptions": {
     "outDir": "./dist",
-    "rootDir": "./types"
+    "rootDir": "./core-utils"
```

Comment:
`types` folder is created as a part of hardhat typechain command, to encompass zksync specific changes, we have decited to put typechain factories under `typechain` folder instead of `types`, and zksync specific code and artifacts under `zksync` folder, so in future, when we add some other chain/protocol, we can simply create another folder in core-utils/

This is the structure used on this branch, there are no changes to the way evm Typechain or build artifacts are imported elsewhere (sdk or cli).

```
Solidity
├── core-utils (or dist)
│   ├── typechain
│   │   ├── contracts
│   │   ├── @openzeppelin
│   │   ├── ...
│   │   └── index.js
│   ├── zksync
│   │   ├── artifacts
│   │   │   ├── ContractName.json
│   │   │   └── ...
│   │   ├── index.ts (zksync specific func)
│   │   └── buildArtifact.js
│   ├── buildArtifact.js (evm)
│   └── index.js (exports)
```

Reply:
the annoying part is switching between git branches that have artifacts in a different path
it breaks the build and its pretty hard to clean up

Reply:
had to manually rm some stuff